e2e/plugin-e2e/canvas/canvas-scene.spec.ts

```diff
@@ -0,0 +1,170 @@
+          // Get element bounds to check if it's outside viewport
+          let canvasSceneWrapper = page.locator('[data-testid="canvas-scene-wrapper"]');
+          let elementBounds = await canvasElement.boundingBox();
+          let viewportBounds = await canvasSceneWrapper.boundingBox();
+
+          if (elementBounds && viewportBounds) {
+            // Check if element is outside the viewport bounds
+            const isOutsideViewport =
+              elementBounds.x + elementBounds.width < viewportBounds.x ||
+              elementBounds.x > viewportBounds.x + viewportBounds.width ||
+              elementBounds.y + elementBounds.height < viewportBounds.y ||
+              elementBounds.y > viewportBounds.y + viewportBounds.height;
+
+            expect(isOutsideViewport).toBe(true);
+          }
+
+          // Test zoom reset with double-click
+          await page.mouse.dblclick(startX, startY);
+          // Verify canvas element is visible after pan/zoom operations
+          await expect(canvasElement).toBeVisible();
+
+          // Test zoom functionality
+          await page.mouse.move(startX, startY);
+          await page.keyboard.down('Control');
+          await page.mouse.wheel(0, -400); // Zoom in
+          await page.keyboard.up('Control');
+
+          // Check if canvas element is not visible after pan/zoom operations
+          canvasElement = await page.getByRole('button', { name: 'Double click to set field' });
+          // Check if element exists in DOM but is outside viewport
+          await expect(canvasElement).toBeAttached(); // Element should still exist in DOM
+
+          // Get element bounds to check if it's outside viewport
+          canvasSceneWrapper = page.locator('[data-testid="canvas-scene-wrapper"]');
+          elementBounds = await canvasElement.boundingBox();
+          viewportBounds = await canvasSceneWrapper.boundingBox();
+
+          if (elementBounds && viewportBounds) {
+            // Check if element is outside the viewport bounds
+            const isOutsideViewport =
+              elementBounds.x + elementBounds.width < viewportBounds.x ||
+              elementBounds.x > viewportBounds.x + viewportBounds.width ||
+              elementBounds.y + elementBounds.height < viewportBounds.y ||
+              elementBounds.y > viewportBounds.y + viewportBounds.height;
+
+            expect(isOutsideViewport).toBe(true);
+          }
+
+          // Test zoom reset with double-click
+          await page.mouse.dblclick(startX, startY);
+          // Verify canvas element is visible after pan/zoom operations
+          await expect(canvasElement).toBeVisible();
+        }
+      }
+    }
+  });
+});
+
+async function createCanvasPanel(page: Page): Promise<void> {
+  // Navigate to Grafana
+  await page.goto('/dashboards');
+
+  // Check if we're on a login page and fill form only if needed
+  try {
```

Comment:
Related to the previous review comment, it'd be nice to do away with the `try/catch` here if we can? This might be one of the exceptions though. 🤔 💭 

In any case, I'll see if we can leverage the existing Grafana E2E tools get this logic out of our way so we don't need to implement it.